### DISCUSSION TITLE
Stroke Color Variance

### DISCUSSION THREAD

**Discussion by andybak** [icosa-gallery/open-brush#331] [New and Experimental Features]
Strokes should be able to vary color over their length.

This will enable selective recolouring of existing strokes, as well as interesting new brush features.

I've started some exploratory coding on this. It is fairly straightforward for GeometryBrush subclasses and trickier for older brush scripts such as QuadStrip etc.

The rough idea is to add a CalcColor method to BaseBrushScript which accepts the main brush color m_Color and a ControlPoint. Adding a color property to ControlPoint and initialise it to 0, 0, 0, 0.

Color mixing is done as a simple color Lerp using the alpha so 0,0,0,0 gives us an initial noop.

As the control point color is taking into account during mesh generation, this shouldn't need special support in other apps as the colors will be correctly baked in during export.

**Comment by billyquith** [comment]
You could do a palette lookup by lerping the sample across a palette strip. if you alter the palette you could animate the scene, or change the tone of it (spring, autumn, winter, psycadelic, underwater, etc).
